src/ln/channelmanager.rs

```diff
@@ -955,7 +952,7 @@ impl ChannelManager {
 					if !chan.is_live() {
 						Some(("Forwarding channel is not in a ready state.", 0x1000 | 7, self.get_channel_update(chan).unwrap()))
 					} else {
-						let fee = amt_to_forward.checked_mul(self.fee_proportional_millionths as u64).and_then(|prop_fee| { (prop_fee / 1000000).checked_add(chan.get_our_fee_base_msat(&*self.fee_estimator) as u64) });
+						let fee = amt_to_forward.checked_mul(self.configuration.channel_options.fee_proportional_millionths as u64).and_then(|prop_fee| { (prop_fee / 1000000).checked_add(chan.get_our_fee_base_msat(&*self.fee_estimator) as u64) });
```

Comment:
This is inconsistent - we have a per-channel fee_proportional_millionths stored in the per-Channel ChannelOptions struct but we're using the default/global copy for fee comparison - one of the two needs to go.

Either the config in ChannelManager should be a default that applies to inbound channels (and new_outbound takes a new one) or we shouldn't have one in Channel at all.

Reply:
The idea is that the per-channel fee_proportional_millionths in config is the default for us and that a new outbound channel is created with the default value until we receive a channel update from the peer telling us what his fee is. 

Looking for that I cannot find where we update that fee in the channel, only in router. But I am going to assume that this should point to the channel specific fee and not default config one?

Reply:
Yea, if we're gonna move to a model where we have per-channel feerates (which should probably be better documented) then this clearly should be checking against that, not the global default one, same with other places in ChannelManager eg ChannelUpdate generation.